catalyst/experiments/hydra_config.py

```diff
@@ -0,0 +1,552 @@
+from typing import Any, Dict, List, Mapping
```

Comment:
dear @otherman16 ,
could we move all duplicated from `ConfigExperiment` to `catalyst/experiments/functional.py`?
or, is it possible to do something like `HydraConfigExperiment(ConfigExperiment)`?

Reply:
I can only move functions `process_callbacks()` and `add_default_callbacks()` to `catalyst/experiments/functional.py`.
The rest of the code are only almost similar.